env/build_test.go

```diff
@@ -19,51 +21,113 @@ func TestBuildEnv(t *testing.T) {
-				"HTTPS_PROXY=some-https-proxy",
-				"https_proxy=some-https-proxy",
-				"HTTP_PROXY=some-http-proxy",
-				"http_proxy=some-http-proxy",
-				"NO_PROXY=some-no-proxy",
-				"no_proxy=some-no-proxy",
-				"NOT_INCLUDED=not-included",
-				"PATH=some-path",
-				"LD_LIBRARY_PATH=some-ld-library-path",
-				"LIBRARY_PATH=some-library-path",
-				"CPATH=some-cpath",
-				"PKG_CONFIG_PATH=some-pkg-config-path",
+		when("platform API >= 0.8", func() {
+			var platformAPI *api.Version
+			it.Before(func() {
+				platformAPI = api.MustParse("0.8")
 			})
-			out := benv.List()
-			sort.Strings(out)
-			expectedVars := []string{
-				"CNB_STACK_ID=some-stack-id",
-				"CPATH=some-cpath",
-				"HOME=some-home",
-				"HOSTNAME=some-hostname",
-				"HTTPS_PROXY=some-https-proxy",
-				"HTTP_PROXY=some-http-proxy",
-				"LD_LIBRARY_PATH=some-ld-library-path",
-				"LIBRARY_PATH=some-library-path",
-				"NO_PROXY=some-no-proxy",
-				"PATH=some-path",
-				"PKG_CONFIG_PATH=some-pkg-config-path",
-			}
-			// Environment variables in Windows are case insensitive, and are added by the lifecycle in uppercase.
-			if runtime.GOOS != "windows" {
-				expectedVars = append(
-					expectedVars,
+			it("can include CNB_ASSETS env var", func() {
+				benv := env.NewBuildEnv([]string{
+					"CNB_STACK_ID=some-stack-id",
+					"HOSTNAME=some-hostname",
+					"HOME=some-home",
+					"HTTPS_PROXY=some-https-proxy",
+					"https_proxy=some-https-proxy",
+					"HTTP_PROXY=some-http-proxy",
 					"http_proxy=some-http-proxy",
+					"NO_PROXY=some-no-proxy",
+					"no_proxy=some-no-proxy",
+					"NOT_INCLUDED=not-included",
+					"PATH=some-path",
+					"LD_LIBRARY_PATH=some-ld-library-path",
+					"LIBRARY_PATH=some-library-path",
+					"CPATH=some-cpath",
+					"PKG_CONFIG_PATH=some-pkg-config-path",
+					"CNB_ASSETS=some-assets-path",
+				}, platformAPI)
+				out := benv.List()
+				sort.Strings(out)
+				expectedVars := []string{
+					"CNB_ASSETS=some-assets-path",
+					"CNB_STACK_ID=some-stack-id",
+					"CPATH=some-cpath",
+					"HOME=some-home",
+					"HOSTNAME=some-hostname",
+					"HTTPS_PROXY=some-https-proxy",
+					"HTTP_PROXY=some-http-proxy",
+					"LD_LIBRARY_PATH=some-ld-library-path",
+					"LIBRARY_PATH=some-library-path",
+					"NO_PROXY=some-no-proxy",
+					"PATH=some-path",
+					"PKG_CONFIG_PATH=some-pkg-config-path",
+				}
+				// Environment variables in Windows are case insensitive, and are added by the lifecycle in uppercase.
+				if runtime.GOOS != "windows" {
+					expectedVars = append(
+						expectedVars,
+						"http_proxy=some-http-proxy",
+						"https_proxy=some-https-proxy",
+						"no_proxy=some-no-proxy",
+					)
+				}
+				if s := cmp.Diff(out, expectedVars); s != "" {
+					t.Fatalf("Unexpected env\n%s\n", s)
+				}
+			})
+		})
+		when("platform API <0.8", func() {
+			var platformAPI *api.Version
+			it.Before(func() {
+				platformAPI = api.MustParse("0.7")
+			})
+			it("includes expected vars", func() {
```

Comment:
```suggestion
			it("does not include CNB_ASSETS env var", func() {
```
(with same comment as https://github.com/buildpacks/lifecycle/pull/591/files?diff=split&w=1#r625334442 )